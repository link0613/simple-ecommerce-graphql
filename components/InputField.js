import React, { PureComponent } from "react";
import { debounce } from "lodash";

class InputField extends PureComponent {
  debounceEvent(...args) {
    this.debouncedEvent = debounce(...args);
    return e => {
      e.persist();
      return this.debouncedEvent(e);
    };
  }

  componentWillUmount() {
    this.debouncedEvent.cancel();
  }

  onChangeText = event => {
    const { value: text } = event.target;  
    const { onChangeText, name } = this.props;
    onChangeText(name, text);
  };

  render() {
    const { placeholder = "", value = "", type = "text" } = this.props;
    return (
      <input
        type={type}
        defaultValue={value}
        onChange={this.debounceEvent(this.onChangeText, 500)}
        placeholder={placeholder}
      />
    );
  }
}

export default InputField;