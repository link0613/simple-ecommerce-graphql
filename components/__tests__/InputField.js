/* eslint-env jest */

import React from "react";
import renderer from "react-test-renderer";

import InputField from "../InputField.js";

describe("Components::<InputField />", () => {
  it("should match snapshot", () => {
    const component = renderer.create(<InputField />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
