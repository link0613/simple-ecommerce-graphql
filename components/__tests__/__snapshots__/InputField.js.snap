// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components::<InputField /> should match snapshot 1`] = `
<input
  defaultValue=""
  onChange={[Function]}
  placeholder=""
  type="text"
/>
`;
